### PR TITLE
CLN: cleanup libs cimports, remove is_timestamp

### PR DESCRIPTION
**Pull request by jbrockmendel**
- cleanup imports in `_libs.parsers`, including some unused
- a couple of functions from util are used exactly once and only in parsers.  Get those directly from numpy_helper instead of from util
- remove unused `util.is_contiguous`
- remove a bunch of distutils shebang-like declarations that are unnecessary and are a pain for when we _do_ want to run tests with coverage enabled.
- remove `tslib.is_timestamp` (will be helpful for eventual goal of "fixing" Timestamp classmethods)
- remove some leftover constants from tslib, update imports appropriately

### CODE REVIEW

File: pandas/tests/test_resample.py [hunk 1]
```diff
@@ -22,8 +22,7 @@
 from pandas.core.base import SpecificationError, AbstractMethodError
 from pandas.errors import UnsupportedFunctionCall
 from pandas.core.groupby import DataError
-from pandas._libs.tslibs.resolution import DAYS
-from pandas.tseries.frequencies import MONTHS
+from pandas._libs.tslibs.resolution import DAYS, _MONTHS as MONTHS
```
Comment (jreback) [inline pandas/tests/test_resample.py]: should fix this this naming mistmacth in resolution (add to list)

File: pandas/tseries/frequencies.py [hunk 2]
```diff
@@ -26,7 +26,7 @@
 from pandas._libs.tslibs.resolution import (Resolution,
                                             _FrequencyInferer,
                                             _TimedeltaFrequencyInferer)
-from pandas._libs.tslibs.parsing import _get_rule_month
+from pandas._libs.tslibs.parsing import _get_rule_month, _MONTH_NUMBERS
```
Comment (jreback) [inline pandas/tseries/frequencies.py]: same here, you can de-privatize these constants

**Review by jreback — APPROVED** [review]
looks fine. some comments to add to the list (or you can fix here). ping on green.

File: pandas/tseries/frequencies.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tseries/frequencies.py]: I think the game plan is to consolidate the several versions of these in ccalendar once it is merged.

**Comment by jreback** [comment]
rebase and something failing

File: pandas/_libs/parsers.pyx [hunk 3]
```diff
@@ -96,6 +81,11 @@ try:
 except NameError:
     basestring = str
 
+cdef extern from "src/numpy_helper.h":
+    object sarr_from_data(cnp.dtype, int length, void* data)
+    void transfer_object_column(char *dst, char *src, size_t stride,
```
Comment (jreback) [inline pandas/_libs/parsers.pyx]: @chris-b1, @gfyoung don't we have an issue to remove support for structured arrays entirely? is that already deprecated?

Comment (jreback) [reply pandas/_libs/parsers.pyx]: if not let's create one (and deprecate in 0.22)

**Review by jreback — CHANGES_REQUESTED** [review]
rebase (and you can now use ccalendar)

**Comment by jbrockmendel** [comment]
Huh the appveyor error is at build-time complaining "unresolved external symbol strcasecmp".  But that particular cimport isn't changed.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**